plugins/image-source/obs-slideshow.c

```diff
@@ -343,21 +425,161 @@ static void ss_update(void *data, obs_data_t *settings)
 			OBS_TRANSITION_SCALE_ASPECT);
 
 	if (ss->randomize && ss->files.num)
-		ss->cur_item = random_file(ss);
+		ss->cur_item = (int)random_file(ss);
 	if (new_tr)
 		obs_source_add_active_child(ss->source, new_tr);
 	if (ss->files.num)
-		obs_transition_start(ss->transition, OBS_TRANSITION_MODE_AUTO,
-				ss->tr_speed,
-				ss->files.array[ss->cur_item].source);
+		do_transition(ss, false);
 
 	obs_data_array_release(array);
 }
 
+static void next_slide(void *data)
+{
+	struct slideshow *ss = data;
+
+	if (++ss->cur_item >= (int)ss->files.num) {
```

Comment:
Because the [style guide](https://github.com/torvalds/linux/blob/master/Documentation/process/coding-style.rst) linked to from this project's [CONTRIBUTING document](https://github.com/jp9000/obs-studio/blob/master/CONTRIBUTING.rst) specifically says so:

> Do not unnecessarily use braces where a single statement will do.